OKR.md

```diff
@@ -2,6 +2,30 @@
 
 We try to frame our ongoing work using a process based on quarterly Objectives and Key Results (OKRs). Objectives reflect outcomes that are challenging, but realistic. Results are tangible and measurable.
 
+## 2018 Q3
+
+**The Connectivity Magic story is figured out and solved**
+
+- A js-ipfs Daemon will always find other IPFS Daemons (js & go) in the LAN 
+- The rendezvous protocol is a sound solution for DApps (e.g. PeerPad & others)
+- The DHT is enabled by default in js-ipfs
+- Handles correctly a scenario where the number of Connections grows (through Conn Management Strategy)
+- ... MOAR :D
+
+**Daemon is as reliable as it can be**
+
+- A Daemon runs for 10 days without a crash
+- One or more js-ipfs daemons are part of the Bootsrapper nodes 
+- js-ipfs can handle 1TB of data, both Node.js and Browser.
+- npm on IPFS over js-ipfs is a sound way to install deps and gets used for CI
+- ... MOAR :D
+
+**New user/contributor experience is extraordinary**
```

Comment:
@hugomrdias I've a few questions based on your suggestions:

- Why codesandbox.io and not codepen?
- Why is it hard to use bundlers with js-ipfs? More than the two examples we have already (browserify + webpack)?

Reply:
codesandbox gives the developer an environment very similar to a local dev environment but inside a browser. You can require/import, bundle, lint, etc. It has integration with github and deploy providers like Now. Even the UI/UX is very similar to popular editors. 
This enables a developer to quickly go from a prototype to a github repo ready to build the next awesome dapp in secs, and the other way around enables us to build examples on github synchronised to a static url on codesandbox from where users can fork and play around. 
some references: 
- https://hackernoon.com/the-future-of-codesandbox-d1ccc5aebf59
- https://twitter.com/mlthuret/status/1012266345529073664
- https://twitter.com/CompuIves/status/1012396990800564225

About bundlers, it's not about being hard to use, it's more about being as easy as possible. Both webpack and parcel provide zero config functionality and we need to make sure ipfs is supported by those default configs. Another thing is, the community is starting to run their bundlers through the complete dependency tree and not use the pre-bundled files from packages, we need to make sure ppl can run webpack/babel through ipfs complete deps tree and still work without errors. An example of this was the work i did to make ipfs work with create-react-app 2.0.

Another topic is size, do we have the best require patterns to enable the best bundle sizes from these bundlers? ex. https://github.com/ipfs/js-ipfs/pull/1414 
Do all of these bundlers include node api polyfills that work as expected ?